src/Compilers/Core/Portable/SymbolDisplay/AbstractSymbolDisplayVisitor.cs

```diff
@@ -75,6 +68,53 @@ protected AbstractSymbolDisplayVisitor NotFirstVisitorNamespaceOrType
             }
         }
 
+        protected virtual void Allocate(
```

Comment:
>virtual [](http://example.com/codeflow?start=18&length=7)

I am not sure whether making this method virtual gives us any advantages. In fact, in combination with name, this somewhat confused me. I think "Initialize" is the name that better reflects the purpose of this API (and similar APIs in derived types). Also, it looks like regular shadowing instead of overriding is going to work and still will allow derived types to call this method.

Reply:
What I _really_ wanted here was to make `Allocate` non-virtual, and then allow a derived type to make its own `Allocate` method with different parameters. The problem I ran into is it made the base class version of `Allocate` trivially callable (the derived method was treated as an overload where both methods were callable, instead of hiding it by name where only the derived one was callable without further qualification). I felt `virtual` made it more difficult to call the wrong method accidentally.

I have no objection to renaming the method to `Initialize` but want to check on if it's acceptable to keep `virtual` for these reasons.

Reply:
➡️ Now updated (method renamed and `virtual` removed)